### PR TITLE
[Google Drive] Add connector documentation

### PR DESCRIPTION
**Pull request by jedrazb**
## Closes https://github.com/elastic/enterprise-search-team/issues/5147

Add Google Drive connector documentation. Note I initially merged #1230 but re-creating it to collect more feedback from Liam and Pius. 

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] this PR has a thorough description

### CODE REVIEW

File: docs/reference/google_drive.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+**NOTE**:
+- Files bigger than 10 MB won't be extracted
+- Permissions are not synced. **All documents** indexed to an Elastic deployment will be visible to **all users with access** to that Elastic Deployment.
+
+## Sync rules
+
+[Basic sync rules](https://www.elastic.co/guide/en/enterprise-search/8.9/sync-rules.html#sync-rules-basic "Basic sync rules") are identical for all connectors and are available by default.
+
+Advanced sync rules are not available for this connector in the present version. Currently filtering is controlled via ingest pipelines.
+
```
Comment (leemthompo) [inline docs/reference/google_drive.md:60]: We usually have a Deployment with Docker section for these connectors too

File: docs/reference/google_drive.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  - In the Cloud Console, go to `IAM and Admin` > `Service accounts` page.
+  - Click the email address of the service account that you want to create a key for.
+  - Click the `Keys` tab. Click the `Add key` drop-down menu, then select `Create new key`.
+  - Select JSON as the Key type and then click `Create`. This will download a JSON file that will contain the service account credentials.
+
+5. **Share Google Drive Folders.** Go to your Google Drive. Right-click the folder or shared drive, choose `Share` and add the email address of the service account you created in step 3. as a viewer to this folder.
+
+  Note: When you grant a service account access to a specific folder or shared drive in Google Drive, it's important to note that the permissions extend to all the children within that folder or drive. This means that any folders or files contained within the granted folder or drive inherit the same access privileges as the parent.
+
+## Configuration
```
[the remark this answers is not in the document]
Comment (jedrazb) [reply docs/reference/google_drive.md:39]: Confirming that `service account JSON` is all we need, similar to https://www.elastic.co/guide/en/enterprise-search/8.9/connectors-google-cloud.html#connectors-google-cloud-configuration

File: docs/reference/google_drive.md (hunk 1 above)
Comment (jedrazb) [reply docs/reference/google_drive.md:119]: Thank you for spotting this, I added the section about Docker

**Review by leemthompo — APPROVED** [review]
Looks good from docs perspective thanks!

**Pull request merged**